pkg/serving/config_changes.go

```diff
@@ -31,14 +32,13 @@ var UserImageAnnotationKey = "client.knative.dev/user-image"
 
 // UpdateEnvVars gives the configuration all the env var values listed in the given map of
 // vars.  Does not touch any environment variables not mentioned, but it can add
-// new env vars and change the values of existing ones.
-func UpdateEnvVars(template *servingv1alpha1.RevisionTemplateSpec, toUpdate map[string]string, toRemove []string) error {
+// new env vars and change the values of existing ones, then sort by environment key name.
+func UpdateEnvVars(template *servingv1alpha1.RevisionTemplateSpec, toUpdate map[string]string, toRemove map[string]bool) error {
 	container, err := ContainerOfRevisionTemplate(template)
 	if err != nil {
 		return err
 	}
-	envVars := updateEnvVarsFromMap(container.Env, toUpdate)
-	container.Env = removeEnvVars(envVars, toRemove)
+	container.Env = mergeEnvVarsWithStableOrder(container.Env, toUpdate, toRemove)
```

Comment:
I'd rather keep the structure where we update them, we remove some, and then we sort them at the end.  Was there a reason you didn't structure it that way?

Reply:
I put together the steps to update and remove EnvVars because I feel it meaningless to append them into a slice and then remove elements from the same slice by using this [slice trick](https://github.com/golang/go/wiki/SliceTricks#delete). I think it's better to handle by appending items unless found in items to remove.

Otherwise, I cut out all the steps into mergeEnvVarsWithStableOrder function for testing purpose. I would like to create a dedicated unit test for detecting the order violation of EnvVars because we have never noticed even this kind of small thing.

Does it make sense?

Reply:
Talking to Navid in Slack, I will only add the ordering logic and keep the structure of updating and removing items as suggested by @sixolet in this PR.